lib/vssh/libssh2.c

```diff
@@ -438,17 +438,35 @@ static int sshkeycallback(struct Curl_easy *easy,
 
 static CURLcode ssh_knownhost(struct Curl_easy *data)
 {
+  int keytype = 0;
+  size_t keylen = 0;
+  int rc = 0;
   CURLcode result = CURLE_OK;
 
+  if(data->set.ssh_hostkeycheck_func) {
+    struct connectdata *conn = data->conn;
+    struct ssh_conn *sshc = &conn->proto.sshc;
+    /* we handle the process to the callback*/
+    const char *remotekey = libssh2_session_hostkey(sshc->ssh_session,
+                                                    &keylen, &keytype);
+
+    if(remotekey) {
```

Comment:
What if there's no remotekey returned? Isn't that then an error?

Reply:
ok, corrected. done